src/dataController.java

```diff
@@ -228,4 +228,32 @@ void deleteMember (memberData obj)
     {
         members.remove(obj);
     }
+
+    //adds a provider object into dataController's arrayList
+    void addProvider(providerData newProvider) {
+        providers.add(newProvider);
+    }
+
+    //returns a provider inside of dataController from user input
+    providerData selectProvider(String providerName) {
+        providerData find = new providerData();
+        for (providerData provider : providers) {
+            if (provider.isMatch(providerName) == true) {
+                find = provider;
+            }
+        }
+        return find;
```

Comment:
What happens when there is no match to a provider?

Reply:
I will work on it!

Reply:
Consider changing the input from string to a scanner, and putting the function inside of an Interface.start() option to test it.
You could force the user to keep putting in input until a successful match is found, and then return the match.